### ISSUE TITLE
AWS CloudTrail Insights

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#10988]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

CloudTrail Insights is designed to automatically analyze management events from your CloudTrail trails to establish a baseline for normal behavior, and then raise issues by generating Insights events when it detects unusual patterns.

### New or Affected Resource(s)

* [`aws_cloudtrail`](https://www.terraform.io/docs/providers/aws/r/cloudtrail.html)

### Potential Terraform Configuration

```hcl
resource "aws_cloudtrail" "example" {
  insight_selector {
    insight_type = "ApiCallRateInsight"
  }
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2019/11/aws-cloudtrail-announces-cloudtrail-insights/).
[Blog post](https://aws.amazon.com/blogs/aws/announcing-cloudtrail-insights-identify-and-respond-to-unusual-api-activity/).
[User guide](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/logging-insights-events-with-cloudtrail.html).

Requires [AWS SDK v1.25.39](https://github.com/aws/aws-sdk-go/releases/tag/v1.25.39):
* https://github.com/terraform-providers/terraform-provider-aws/pull/10955

**Comment by omuthu-personal** [comment]
Any ETA on PR #12390

**Comment by bflad** [comment]
Support for this functionality has been merged and will release with version 3.5.0 of the Terraform AWS Provider, likely tomorrow. Thanks to @DrFaust92 for the implementation. 👍
